runtime/src/main/java/io/quarkiverse/logging/splunk/SplunkLogHandlerRecorder.java

```diff
@@ -91,6 +92,17 @@ static HttpEventCollectorSender createSender(SplunkHandlerConfig config) {
             sender.setEventHeaderSerializer(serializer);
             sender.setEventBodySerializer(serializer);
         }
+        if (config.middleware.isPresent()) {
+            try {
+                sender.addMiddleware(
+                        Thread.currentThread().getContextClassLoader().loadClass(config.middleware.get())
+                                .asSubclass(HttpSenderMiddleware.class)
+                                .getDeclaredConstructor()
+                                .newInstance());
```

Comment:
I am bit sceptical this is gonna work with native compilation : The new instance of the middleware is done by reflection which is not supported by native image without the proper metadata: native image need to known all classes used at runtime otherwise it tries to remove every class definition not used int the final executable.

So we need to instrument the extension to retain the class definition, for that we need to produce a ReflectiveClassBuildItem from the LoggingSplunkProcessor: this might require a build time config which makes sense because the middleware is known at build time.
I would also provide a test in the `integration-test` package: we are performing native compilation there to test the extension.